src/agent/src/rpc.rs

```diff
@@ -1571,15 +1573,19 @@ fn append_guest_hooks(s: &Sandbox, oci: &mut Spec) -> Result<()> {
 
 // Check is the container process installed the
 // handler for specific signal.
-fn is_signal_handled(pid: pid_t, signum: u32) -> bool {
+fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
+    if signum < 1 {
```

Comment:
signum 0 is a special number used to check a process liveness. Do we want to keep it being handled?

signum 1 is SIGHUP. we should handle it too IMO.

Reply:
@bergwolf 
> signum 1 is SIGHUP. we should handle it too IMO.

I think signum 1 is still be handled properly. If it is `1`, the `sig_mask` variable is set to `1`, so it will be able to be compared with the process's `SigCgt` field. Please let me know if I'm misunderstanding.

> signum 0 is a special number used to check a process liveness. Do we want to keep it being handled?

I'm not sure how we would check that using the `SigCgt` field. Would it make sense to just return true if the process status file exists and skip the comparison with the `SigCgt` field in the case `signum == 0`?

Reply:
Yep - I think this is fine as we're not stopping signal zero being delivered, we're just checking the signal mask here.